xbmc/addons/Addon.cpp

```diff
@@ -391,6 +391,30 @@ AddonVersion CAddon::GetDependencyVersion(const std::string &dependencyID) const
   return AddonVersion("0.0.0");
 }
 
+int CAddon::NoDeps(std::map<std::string,std::string>& added) const
+{
+  int result = 0;
+  for (const auto& it : GetDeps())
+  {
+    if (StringUtils::StartsWith(it.first, "xbmc.") ||
+        StringUtils::StartsWith(it.first, "kodi."))
+      continue;
+    if (added.find(it.first) != added.end())
+    {
+      if (!added[it.first].empty() &&
+          AddonVersion(it.second.first.asString()) < AddonVersion(added[it.first]))
+        continue;
+    }
+    added[it.first] = it.second.first.asString();
+    AddonPtr addon;
+    result++;
+    if (CServiceBroker::GetAddonMgr().FindInstallableById(it.first, addon))
```

Comment:
ehm, disregard. i was working on removing the recursion and my head fooled me into thinking you were on about the same.